registrations/filters.py

```diff
@@ -0,0 +1,193 @@
+        registrations = registrations.filter(qs_filter)
+
+        if not registrations:
+            raise DRFPermissionDenied(
+                _(
+                    "Only the admins of the registration organizations have access rights."
+                )
+            )
+
+        return queryset.filter(registration__in=registrations)
+
+    @staticmethod
+    def _build_text_annotations(relation_accessor=""):
+        return {
+            f"{relation_accessor}first_last_name": Concat(
+                f"{relation_accessor}first_name",
+                Value(" "),
+                f"{relation_accessor}last_name",
+            ),
+            f"{relation_accessor}last_first_name": Concat(
+                f"{relation_accessor}last_name",
+                Value(" "),
+                f"{relation_accessor}first_name",
+            ),
+        }
+
+    @staticmethod
+    def _build_text_filter(text_param, relation_accessor=""):
+        filters = {
+            f"{relation_accessor}first_last_name__icontains": text_param,
+            f"{relation_accessor}last_first_name__icontains": text_param,
+            "contact_person__email__icontains": text_param,
+            "contact_person__membership_number__icontains": text_param,
+            "contact_person__phone_number__icontains": text_param,
+        }
+
+        q_set = Q()
+        for item in filters.items():
+            q_set |= Q(item)
+
+        return q_set
+
+    def filter_text(self, queryset, name, value):
+        return queryset.annotate(**self._build_text_annotations()).filter(
+            self._build_text_filter(value)
+        )
+
+
+class SignUpFilter(SignUpBaseFilter):
+    class Meta:
+        model = SignUp
+        fields = ("registration", "attendee_status")
+
+
+class SignUpGroupFilter(SignUpBaseFilter):
+    attendee_status = django_filters.MultipleChoiceFilter(
+        field_name="signups__attendee_status",
+        choices=SignUp.ATTENDEE_STATUSES,
+        widget=django_filters.widgets.CSVWidget(),
+    )
+
+    def filter_text(self, queryset, name, value):
+        return queryset.annotate(
+            **self._build_text_annotations(relation_accessor="signups__")
+        ).filter(self._build_text_filter(value, relation_accessor="signups__"))
+
+    class Meta:
+        model = SignUpGroup
+        fields = ("registration",)
+
+
+class PriceGroupFilter(ActionDependingFilter):
+    publisher = django_filters.Filter(
+        widget=django_filters.widgets.CSVWidget(),
+        method="filter_publisher",
+    )
+
+    def filter_publisher(self, queryset, name, value: list[str]):
+        if value == ["None"] or value == ["none"]:
+            return queryset.filter(publisher=None)
+
+        publishers = Organization.objects.filter(pk__in=value).values_list(
+            "pk", flat=True
+        )
+        if not publishers:
+            return self.Meta.model.objects.none()
+
+        return queryset.filter(publisher_id__in=publishers)
```

Comment:
It would be nice to filter default and publisher specific price group with a single request e.g. `/price_group/?publisher=ahjo:00001,none`

At the moment `/price_group/?publisher=ahjo:00001,ahjo:u021600` and `/price_group/?publisher=none` works as expected but price groups without publisher are excluded when trying to use `/price_group/?publisher=ahjo:00001,none` filter

Reply:
Filtering by default and specific publishers at the same time hows now been implemented.